RESTserver/src/main/java/com/softserve/if072/restservice/service/ShoppingListService.java

```diff
@@ -39,7 +39,12 @@ public void insert(ShoppingList shoppingList) {
     }
 
     public void update(ShoppingList shoppingList) throws DataNotFoundException {
-        shoppingListDAO.update(shoppingList);
+        if ((shoppingList.getAmount() <= 0) || (shoppingList.getProduct() == null)
+                || (shoppingList.getUser() == null)) {
+            throw new DataNotFoundException("Incorrect fields by ShoppingList");
```

Comment:
For this case , better to thow IllegalArgumentException. Because DataNotFoundException is used for the cases when data is not found in the DB